crypto/addresses/src/bech32.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{Address, AddressError, Prefix};
+
+const CHARSET: &[u8] = b"qpzry9x8gf2tvdw0s3jn54khce6mua7l";
+const REV_CHARSET: [u8; 123] = [
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 15, 100, 10, 17, 21, 20, 26, 30, 7, 5, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100, 100,
+    100, 100, 100, 29, 100, 24, 13, 25, 9, 8, 23, 100, 18, 22, 31, 27, 19, 100, 1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2,
+];
+
+// Checksome for bech32
+// https://bch.info/en/specifications
+fn polymod<'data, I>(values: I) -> u64
+where
+    I: Iterator<Item = &'data u8>,
```

Comment:
Can you make this `Iterator<Item=u8>`? a reference is bigger than a u8(and it is a copy type) so there's no need to pass references

Reply:
Done